### ISSUE TITLE
Mishandling of lu on view of SparseMatrixCSC

### ISSUE THREAD

**Issue by vtjnash** [JuliaSparse/SparseArrays.jl#436]
MWE:
```julia-repl
julia> A = sparse(rand(5,5));

julia> lu(view(A,:,:))
ERROR: MethodError: no method matching lu!(::SparseMatrixCSC{Float64, Int64}, ::RowMaximum; check=true)
```
EDIT: Julia v1.8.0.

This arose in [this Discourse thread](https://discourse.julialang.org/t/error-methoderror-no-method-matching-lu/91618).

The issue appears to be that `lu(::SubArray)` adds the `pivot` argument to its `lu!` call, which is unsupported by the `SparseMatrixCSC` version. Either the `pivot` argument should be accepted to the sparse version (even if ignored for now) or this call should be more careful about applying it. Or this should hit a generic fallback (but that might be extremely slow).
EDIT: actually it appears to be a bit more complicated because `lu!` for sparse arrays has a rather different signature. Still, the issue lies somewhere around the call chain invoked by `lu(::SubArray)`.

I personally don't advocate for ever ending up in this place, as one should prefer to materialize the view and use ~~`lu!`~~ `lu` (I guess `lu!` for sparse arrays is a bit more complicated to use) on the materialized version. But this is broken in any case and might be indicative of similarly brittle behaviors with other types so is worth resolving.

_Originally posted at https://github.com/JuliaLang/julia/issues/47908 by @mikmoore_

**Comment by vtjnash** [comment]
Seems to be fixed in v1.9, as commented in that issue
```
julia> lu(view(A,:,:))
LU{Float64, SparseMatrixCSC{Float64, Int64}, Vector{Int64}}
L factor:
5×5 SparseMatrixCSC{Float64, Int64} with 15 stored entries:
 1.0        ⋅          ⋅         ⋅       ⋅ 
 0.679268  1.0         ⋅         ⋅       ⋅ 
 0.36618   0.981632   1.0        ⋅       ⋅ 
 0.170215  0.860001  -0.542062  1.0      ⋅ 
 0.444488  0.164982  -0.109861  0.1973  1.0
U factor:
5×5 SparseMatrixCSC{Float64, Int64} with 15 stored entries:
 0.650809  0.30193   0.813484   0.0910626   0.796603
  ⋅        0.580244  0.114827   0.932396    0.0845381
  ⋅         ⋅        0.225394  -0.850349    0.0425463
  ⋅         ⋅         ⋅        -1.21116    -0.0789594
  ⋅         ⋅         ⋅          ⋅          0.0432081
```

**Comment by vtjnash** [comment]
The thing is, it does not really matter as long as it returns a correct answer
